src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -2151,7 +2206,11 @@ def serialize_key_and_certificates_to_pkcs12(
         with self._zeroed_null_terminated_buf(password) as password_buf:
             with self._zeroed_null_terminated_buf(name) as name_buf:
                 ossl_cert = self._cert2ossl(cert) if cert else self._ffi.NULL
-                evp_pkey = key._evp_pkey if key else self._ffi.NULL
+                if key is not None:
+                    evp_pkey = key._evp_pkey  # type: ignore[union-attr]
+                else:
+                    evp_pkey = self._ffi.NULL
```

Comment:
Only the error suggested any of the three was acceptable. I've pushed an update that has a more specific error msg for one case, although the logic is probably more convoluted than it needs to be since I've been doing this too much today 😁

Reply:
Okay, mitmproxy relies on the existing behavior (where passing `cert` without `key` makes it show up in `cas`), so I'll just add tests for this behavior, document it a bit better, and move on.